filebeat/crawler/registrar.go

```diff
@@ -16,7 +17,8 @@ type Registrar struct {
 	// Path to the Registry File
 	registryFile string
 	// Map with all file paths inside and the corresponding state
-	State map[string]*FileState
+	State      map[string]*FileState
+	stateMutex *sync.Mutex
```

Comment:
Seem like there is a mismatch in visibility here. The mutex and the data it protects should both be either exported or unexported.